models/DataObject/ClassDefinition/Data.php

```diff
@@ -740,7 +740,13 @@ public function getSetterCode($class)
         }
 
         if ($this->supportsDirtyDetection()) {
+            if($class instanceof DataObject\ClassDefinition && $class->getAllowInherit()) {
+                $code .= "\t" . "self::setGetInheritedValues(false);\n";
+            }
             $code .= "\t" . '$currentData = $this->get' . ucfirst($this->getName()) . '();' . "\n";
+            if($class instanceof DataObject\ClassDefinition && $class->getAllowInherit()) {
+                $code .= "\t" . "self::setGetInheritedValues(true);\n";
```

Comment:
*) get original state before setting it to false and restore the state here instead of setting it to true
*) same change probably also needed for localized field and brick

Reply:
Fixed.